upstairs/src/lib.rs

```diff
@@ -4397,144 +4876,74 @@ async fn up_listen(
-                    if let Some(c) = c {
-                        println!(
-                            "[{}] {:?} connection:{:?}",
-                            c.client_id, c.target, c.connected,
-                        );
-                        /*
-                         * If this just connected, see if the
-                         * reconciliation can now pass.
-                         */
-                        if c.connected
-                            && up.ds_reconciliation(&dst, &mut lastcast)
-                        {
-                            break;
-                        }
-                    } else {
-                        println!("#### ? #### DISCONNECTED due to None! ####");
-                    }
-                }
-                req = up.guest.recv() => {
+        tokio::select! {
+            c = ds_status_rx.recv() => {
+                if let Some(c) = c {
+                    println!(
+                        "[{}] {:?} new connection:{:?}",
+                        c.client_id, c.target, c.connected,
+                    );
+                    up.ds_state_show();
                     /*
-                     * There are a few commands we will accept before we
-                     * have all downstairs online. Other commands should
-                     * return an error.  process_new_io should be able
-                     * to handle the difference.
+                     * If this just connected, see if the
+                     * reconciliation can now pass.
                      */
-                    process_new_io(up, &dst, req, &mut lastcast).await;
+                    if c.connected {
+                        up.ds_reconciliation(&dst, &mut lastcast);
+                    } else {
+                        println!("[{}] is offline {} ", c.client_id, c.target);
+                    }
+                } else {
+                    // I think this is a panic?  Or exit.. I don't think
+                    // that ds_status_rx should ever go away.
+                    panic!("#### ? #### DISCONNECTED due to None! ####");
                 }
             }
-        }
-
-        stat_update(up, "loop end");
-        println!("All downstairs online, Now accepting IO requests",);
-
-        up.ds_state_show();
-        /*
-         * We have three connections, so we can now start listening for
-         * more IO to come in. We also need to make sure our downstairs
-         * stay connected, and we watch the ds_status_rx.recv() for that
-         * to change which is our notification that a disconnect has
-         * happened.
-         *
-         * In addition, we also send a periodic flush when we determine it
-         * is time to do so. TODO: Figure out when is the best time to send
-         * a upstairs generated flush.
-         */
-        let mut flush_check = deadline_secs(5);
-        let mut show_work_interval = deadline_secs(5);
+            req = up.guest.recv() => {
```

Comment:
it looks like this now processes post-all-downstairs-coming-online guest requests, so this comment should be removed

Reply:
Done in 9608a17